### ISSUE TITLE
net: TestCloseWrite for SOCK_SEQPACKET failing on SmartOS

### ISSUE THREAD

**Issue by bcmills** [golang/go#30840]
Looking at the recent test flakes on `solaris-amd64-smartosbuildlet`, the vast majority of them are failures in `TestCloseWrite`.

For an example, see https://build.golang.org/log/f41d2e44b0629db5f3b5e32eec5751b465f4717f:
```
--- FAIL: TestCloseWrite (0.00s)
    net_test.go:153: got (0, read unixpacket ->/tmp/workdir/tmp/go-nettest264897916: not pollable); want (0, io.EOF)
FAIL
FAIL	net	9.459s
```

CC @mikioh @bradfitz @ianlancetaylor

**Comment by ianlancetaylor** [comment]
Most likely due to https://golang.org/cl/166497 "runtime, internal/poll, net: report event scanning error on read event".  I suspect that Solaris `poll` is reporting `POLLERR` if the other side of a Unix socket is closed, although the man page is uninformative on the point.  It would be interesting to know whether `POLLHUP` was also set.

**Comment by mikioh** [comment]
Sure, I think the fix for #30817 also cures this issue.

**Comment by mikioh** [comment]
Reopen: it seems like there's something-something between illumos and SmartOS.
